crates/iroha_telemetry_derive/src/lib.rs

```diff
@@ -237,9 +245,9 @@ fn impl_metrics(emitter: &mut Emitter, _specs: &MetricSpecs, func: &syn::ItemFn)
     };
 
     #[cfg(feature = "metric-instrumentation")]
-    let res = {
-        let (totals, successes, times) = write_metrics(_metric_arg_ident, _specs);
-        quote!(
+    {
+        let (totals, successes, times) = write_metrics(metric_arg_ident, specs);
+        return quote!(
```

Comment:
```suggestion
       quote!(
```

I don't think you need a return